lib/util/Components.js

```diff
@@ -419,14 +419,81 @@ function componentRule(rule, context) {
       return utils.isReturningJSX(ASTNode, strict) || utils.isReturningNull(ASTNode);
     },
 
+    getComponentNameFromJSXElement(node) {
+      if (node.type !== 'JSXElement') {
+        return null;
+      }
+      if (node.openingElement && node.openingElement.name && node.openingElement.name.name) {
+        return node.openingElement.name.name;
+      }
+      return null;
```

Comment:
when would we expect to reach here, if it's a JSXElement node? An explicit exception would catch parser bugs more quickly.

Reply:
Shorthand expression of React Fragment will be JSXElement without a name